### PR TITLE
Rework dialog boxes to use modern components and sizing

### PR DESCRIPTION
**Pull request by Vagabond**
Dialog boxes were previously very hardcoded. This PR makes them use the more modern container widgets and text rendering so they are more flexible with regard to their contents.

This also improves the menu widget layout behavior and adds support for centered horizontal menus, which are used for the dialog buttons.

Dialog boxes are now made up of a menu containing a textlabel, for the dialog text, then a horizontal, centered menu that holds the dialog buttons.

### CODE REVIEW

**Comment by Vagabond** [comment]
Don't merge this yet, it breaks the mainmenu somewhat.

**Comment by Vagabond** [comment]
Ok, this should be good now.

File: src/game/scenes/vs.c [hunk 1]
```diff
@@ -617,7 +618,7 @@ int vs_create(scene *scene) {
     scene_set_render_cb(scene, vs_render);
     scene_set_render_overlay_cb(scene, vs_render_overlay);
     scene_set_input_poll_cb(scene, vs_input_tick);
-    scene_set_dynamic_tick_cb(scene, vs_dynamic_tick);
+    // scene_set_dynamic_tick_cb(scene, vs_dynamic_tick);
```
Comment (katajakasa) [inline src/game/scenes/vs.c:621]: No longer needed ?

Comment (Vagabond) [reply src/game/scenes/vs.c:621]: it never seemed necessary

Comment (katajakasa) [reply src/game/scenes/vs.c:621]: Ok. Just remove the line and the tick function completely ? We can find it in git history if it's needed.

Comment (Vagabond) [reply src/game/scenes/vs.c]: Fixed, I just forgot to remove it.

**Review by katajakasa — APPROVED** [review]

**Pull request merged**